### PR TITLE
fix: frontend markdown rendering in result panels

### LINKED ISSUE #55: feat(skills): define frontend visualization skills for structural data and results
### Body / 正文

This issue defines frontend visualization skills for structural models and analysis results. These skills should bridge normalized data with rendering tools such as Plotly.js and PyVista-related workflows.

这个 issue 用于定义面向结构模型和计算结果的前端可视化技能。这类技能应在统一数据和 Plotly.js、PyVista 相关渲染流程之间建立桥梁。

### Checklist / 清单

- Define visualization-oriented data contracts
- Clarify responsibilities between backend preprocessing and frontend rendering
- Define extensibility for multiple visualization backends

- 定义面向可视化的数据协议
- 明确后端预处理与前端渲染的职责边界
- 定义面向多种可视化后端的扩展机制

### Done When / 完成标准

- Visualization skills have reusable contracts
- Backend/frontend boundaries are documented
- At least one visualization path is demonstrable

- 可视化技能具备可复用协议
- 前后端职责边界已文档化
- 至少一条可视化路径可以演示

### PR DESCRIPTION
**Pull request by qinsz01**
## Summary

Describe the problem and fix in 2–5 bullets:

If this PR fixes a plugin beta-release blocker, title it `fix(<plugin-id>): beta blocker - <summary>` and link the matching `Beta blocker: <plugin-name> - <summary>` issue labeled `beta-blocker`. Contributors cannot label PRs, so the title is the PR-side signal for maintainers and automation.

- Problem: The frontend only rendered raw text in the execution summary and conversation guidance body fields, and the Markdown report view did not support GFM tables.
- Why it matters: Analysis/reporting results lost formatting, so headings, emphasis, and tables from backend output were hard to read in the console.
- What changed: Added shared Markdown rendering for result-panel body content, enabled `remark-gfm` for table support, and expanded console integration coverage for Markdown rendering and current `StructureModel v2` UI expectations.
- What did NOT change (scope boundary): No backend payload format, agent orchestration, or report-generation logic changed.

## Change Type (select all)

- [x] Bug fix
- [ ] Feature
- [ ] Refactor required for the fix
- [ ] Docs
- [ ] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [ ] Gateway / orchestration
- [ ] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [ ] API / contracts
- [x] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR

- Closes #
- Related #
- [x] This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)

For bug fixes or regressions, explain why this happened, not just what changed. Otherwise write `N/A`. If the cause is unclear, write `Unknown`.

- Root cause: The console used plain text rendering for result-panel body strings, and `react-markdown` was mounted without the GFM plugin, so table syntax stayed raw.
- Missing detection / guardrail: Integration coverage did not assert Markdown rendering for execution/guidance body fields or table rendering in the report panel.
- Prior context (`git blame`, prior PR, issue, or refactor if known): The current UI copy and tests had already moved to `StructureModel v2`, but several older assertions still queried the deprecated `v1` placeholder string.
- Why this regressed now: Result-panel content evolved to carry Markdown-rich text, while the frontend display path stayed on raw string output.
- If unknown, what was ruled out: Ruled out backend payload/schema changes; the issue was isolated to frontend rendering and stale test assertions.

## Regression Test Plan (if applicable)

For bug fixes or regressions, name the smallest reliable test coverage that should have caught this. Otherwise write `N/A`.

- Coverage level that should have caught this:
  - [ ] Unit test
  - [x] Seam / integration test
  - [ ] End-to-end test
  - [ ] Existing coverage already sufficient
- Target test or file: `frontend/tests/integration/console-page.test.tsx`
- Scenario the test should lock in: Execution summary, conversation guidance, and report summary/body should render Markdown formatting, and report tables should render as actual HTML tables.
- Why this is the smallest reliable guardrail: The regression sits at the rendered console surface where translated labels, archived state, and streamed result payloads meet.
- Existing test that already covers this (if any): Existing console-page integration coverage already exercised the same result panel; this PR extends that file with Markdown-specific assertions.
- If no new test is added, why not: N/A

## User-visible / Behavior Changes

- Execution summary body now renders Markdown formatting instead of raw markers.
- Conversation guidance body/support note/next-step text now renders Markdown formatting instead of raw markers.
- Markdown reports now render GFM tables in the frontend console.

## Diagram (if applicable)

For UI changes or non-trivial logic flows, include a small ASCII diagram reviewers can scan quickly. Otherwise write `N/A`.

```text
Before:
[backend markdown text] -> [result panel raw string render] -> [**bold** and |table| syntax shown literally]

After:
[backend markdown text] -> [shared markdown renderer + remark-gfm] -> [formatted prose/table output in console]
```

## Security Impact (required)

- New permissions/capabilities? (`Yes/No`) No
- Secrets/tokens handling changed? (`Yes/No`) No
- New/changed network calls? (`Yes/No`) No
- Command/tool execution surface changed? (`Yes/No`) No
- Data access scope changed? (`Yes/No`) No
- If any `Yes`, explain risk + mitigation:

### Environment

- OS: Linux
- Runtime/container: local Node/Next.js frontend test environment
- Model/provider: N/A
- Integration/channel (if any): frontend Vitest integration tests
- Relevant config (redacted): default local test config

### Steps

1. Return a streamed agent result whose `response`, `interaction`, or `report.summary` contains Markdown markers such as `**bold**`.
2. Return a `report.markdown` payload containing a GFM table.
3. Open the console result panels.

### Expected

- Markdown body fields render formatted text.
- Markdown report tables render as actual tables.

### Actual

- Before this change, body fields displayed raw Markdown markers and tables stayed as plain text.

## Evidence

Attach at least one:

- [x] Failing test/log before + passing after
- [ ] Trace/log snippets
- [ ] Screenshot/recording
- [ ] Perf numbers (if relevant)

## Human Verification (required)

What you personally verified (not just CI), and how:

- Verified scenarios: Ran frontend type-check, targeted Markdown rendering integration tests, and the full `console-page` integration suite.
- Edge cases checked: Report table rendering, archived-state assertions against `StructureModel v2`, and async aborted-stream archive persistence.
- What you did **not** verify: Manual browser interaction outside automated frontend tests.

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

If a bot review conversation is addressed by this PR, resolve that conversation yourself. Do not leave bot review conversation cleanup for maintainers.

## Compatibility / Migration

- Backward compatible? (`Yes/No`) Yes
- Config/env changes? (`Yes/No`) No
- Migration needed? (`Yes/No`) No
- If yes, exact upgrade steps:

## Risks and Mitigations

List only real risks for this PR. Add/remove entries as needed. If none, write `None`.

- Risk: Markdown styling could affect spacing/readability in non-report result cards.
  - Mitigation: Scoped rendering to body text fields only and covered the affected console surface with integration tests.

### CODE REVIEW

File: frontend/src/components/chat/ai-console.tsx [hunk 1]
```diff
@@ -299,6 +300,22 @@ type CapabilityMatrixPayload = {
...
+  className,
+}: {
+  content: string
+  className?: string
+}) {
+  return (
+    <div className={cn(MARKDOWN_BODY_CLASS, className)}>
+      <ReactMarkdown remarkPlugins={[remarkGfm]}>{content}</ReactMarkdown>
+    </div>
+  )
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply frontend/src/components/chat/ai-console.tsx:330]: Checked against the installed `react-markdown` implementation (`frontend/node_modules/react-markdown/lib/index.js`): passing `className` causes the library to wrap the output in a `div` itself. So applying the suggestion would still produce a wrapper `div`, not a cleaner DOM. Keeping the explicit wrapper here makes the container shape and the `prose`/table styling boundary explicit in our component, so I’m leaving the code as-is and resolving this thread.

File: frontend/src/components/chat/ai-console.tsx [hunk 2]
```diff
@@ -299,6 +300,22 @@ type CapabilityMatrixPayload = {
...
+  content,
+  className,
+}: {
+  content: string
+  className?: string
+}) {
+  return (
+    <div className={cn(MARKDOWN_BODY_CLASS, className)}>
+      <ReactMarkdown remarkPlugins={[remarkGfm]}>{content}</ReactMarkdown>
+    </div>
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply frontend/src/components/chat/ai-console.tsx:329]: Fixed in d812286. `MarkdownBody` now rewrites backend-relative markdown URLs through `urlTransform`, so `/api/v1/files/serve?...` links resolve against `API_BASE` instead of the Next.js origin. The anchor renderer also keeps `target="_blank"` and `rel="noopener noreferrer"` aligned with `renderContentWithLinks`. Added an integration test that exercises report markdown links end to end.

File: frontend/src/components/chat/ai-console.tsx [hunk 3]
```diff
@@ -299,6 +300,22 @@ type CapabilityMatrixPayload = {
...
+
+function MarkdownBody({
+  content,
+  className,
+}: {
+  content: string
+  className?: string
+}) {
+  return (
+    <div className={cn(MARKDOWN_BODY_CLASS, className)}>
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply frontend/src/components/chat/ai-console.tsx]: Fixed in d812286. The markdown helper now has a base typography class plus a `compact` variant instead of forcing `prose-p:my-0` everywhere. Execution summary and guidance cards stay compact, while report summary/body render with default paragraph spacing again. Added an integration test that asserts the compact spacing stays in analysis cards and does not leak into the report panel.

**Review by guyi2000 — APPROVED** [review]
